src/components/veNft/veNftConfirmStakeModal.tsx

```diff
@@ -1,28 +1,144 @@
 
-interface ConfirmStakeModalProps {
+import { NetworkContext } from 'contexts/networkContext'
+import { useLockTx } from 'hooks/veNft/transactor/VeNftLockTx'
+
+import { useContext, useState } from 'react'
+import { formattedNum, parseWad } from 'utils/formatNumber'
+
+import { detailedTimeString } from 'utils/formatTime'
+import {
+  emitErrorNotification,
+  emitSuccessNotification,
+} from 'utils/notifications'
+
+type ConfirmStakeModalProps = {
   visible: boolean
-  onCancel: VoidFunction
   tokenSymbolDisplayText: string
+  tokensStaked: number
+  votingPower: number
+  lockDuration: number
+  beneficiary: string
+  onCancel: VoidFunction
+  onCompleted: VoidFunction
 }
 
-const ConfirmStakeModal = ({
+export default function ConfirmStakeModal({
   visible,
-  onCancel,
   tokenSymbolDisplayText,
-}: ConfirmStakeModalProps) => {
+  tokensStaked,
+  votingPower,
+  lockDuration,
+  beneficiary,
+  onCancel,
+  onCompleted,
+}: ConfirmStakeModalProps) {
+  const { userAddress, onSelectWallet } = useContext(NetworkContext)
+  const [loading, setLoading] = useState(false)
+  const recipient = beneficiary !== '' ? beneficiary : userAddress
+
+  const tokensStakedInWad = parseWad(tokensStaked)
+
+  const formattedLockDuration = detailedTimeString({
+    timeSeconds: lockDuration,
+    fullWords: true,
+  })
+
+  const lockTx = useLockTx()
+
+  async function lock() {
+    if (!recipient) {
+      emitErrorNotification(
+        t`No beneficiary selected. Is your wallet connected?`,
+      )
+      return
+    }
+
+    // Prompt wallet connect if no wallet connected
+    if (!userAddress && onSelectWallet) {
+      onSelectWallet()
+    }
+
+    setLoading(true)
+
+    const txSuccess = await lockTx(
+      {
+        account: userAddress!,
+        value: tokensStakedInWad,
+        lockDuration: lockDuration,
+        beneficiary: recipient,
+        useJbToken: true,
+        allowPublicExtension: false,
+      },
+      {
+        onConfirmed() {
+          setLoading(false)
+          emitSuccessNotification(
+            t`Lock successful. Results will be indexed in a few moments.`,
+          )
+          onCompleted()
+        },
+      },
+    )
+
+    if (!txSuccess) {
+      setLoading(false)
+    }
+  }
+
   return (
     <Modal
       visible={visible}
       onCancel={onCancel}
+      onOk={lock}
       okText={`Lock $${tokenSymbolDisplayText}`}
+      confirmLoading={loading}
     >
       <h2>
         <Trans>Confirm Stake</Trans>
       </h2>
+      <Callout>
+        <Trans>
+          You are agreeing to IRREVOCABLY lock your tokens for{' '}
+          {formattedLockDuration} in exchange for {votingPower} $ve
+          {tokenSymbolDisplayText}
+        </Trans>
+      </Callout>
+      <Row>
+        <Col span={14}>
+          <Descriptions
+            title={
+              <h3>
+                <Trans>$ve{tokenSymbolDisplayText} NFT summary:</Trans>
+              </h3>
+            }
```

Comment:
**suggestion** feel like we could remove this. wdyt?

Reply:
SGTM